src/librustc/middle/lang_items.rs

```diff
@@ -417,4 +30,36 @@ impl<'tcx> TyCtxt<'tcx> {
             }
         })
     }
+
+    pub fn fn_trait_lang_item(&self, id: DefId) -> Option<ty::ClosureKind> {
```

Comment:
This should be named `fn_trait_kind_from_lang_item`. It currently sounds like it returns a lang item.